src/main/java/com/github/crimsondawn45/fabricshieldlib/mixin/LivingEntityMixin.java

```diff
@@ -19,40 +21,80 @@
-                    Entity sourceEntity = source.getSource();
-
-                    if (sourceEntity instanceof LivingEntity) {
-                        ((LivingEntityAccessor) entity).fabricshieldlib$invokeTakeShieldHit((LivingEntity) sourceEntity);
-                    }
-                }
-            }
-        }
-    }
-
-    //Makes armor stands equip shields correctly
-    @Inject(at = @At(value = "HEAD"), method = "getPreferredEquipmentSlot(Lnet/minecraft/item/ItemStack;)Lnet/minecraft/entity/EquipmentSlot;", cancellable = true)
-    private void getPreferredEquipmentSlot(ItemStack stack, CallbackInfoReturnable<EquipmentSlot> callbackinfo) {
-
-        if(stack.getItem() instanceof FabricShield) {
-            callbackinfo.setReturnValue(EquipmentSlot.OFFHAND);
-        }
-    }
+public abstract class LivingEntityMixin {
+	@Inject(
+			at = @At(value = "HEAD"),
+			method = "damage(Lnet/minecraft/server/world/ServerWorld;Lnet/minecraft/entity/damage/DamageSource;F)Z"
+		)
+	private void invokeEvent(ServerWorld world, DamageSource source, float amount,
+			CallbackInfoReturnable<Boolean> callbackInfo) {
+		// Handle shield blocking
+		LivingEntity entity = (LivingEntity) (Object) this;
+		ItemStack activeItem = entity.getActiveItem();
+		ShieldBlockCallback.EVENT.invoker().block(entity, source, amount, entity.getActiveHand(), activeItem);
+	}
+
+	/**
+	 * This mixin changes the return value of "itemStack.getItem() instanceof ShieldItem".
+	 *
+	 * @param item        "itemStack.getItem()", the item to be tested
+	 * @param shieldClass always be "ShieldItem.class"
+	 * @return If the item is a shield, return "Object.class" so that "instanceof"
+	 *         always results in true. Otherwise, return "Integer.class". As Item is
+	 *         not assignable from Integer, "instanceof" always results in false.
+	 */
+    /*
+	@ModifyConstant(
+			allow = 1,
+			require = 1,
+			method = "blockedByShield(Lnet/minecraft/entity/damage/DamageSource;)V",
+			slice = @Slice(from = @At(value = "INVOKE",
+				target = "Lnet/minecraft/entity/LivingEntity;getBlockingItem()Lnet/minecraft/item/ItemStack;")),
+			constant = @Constant(classValue = ShieldItem.class)
+		)
+	private Class<?> instanceOfShieldItem(Object item, Class<?> shieldClass) {
+		return FabricShieldUtils.isShieldItem((Item) item) ? Object.class : Integer.class;
+	}
+	*/
+
+    /* I believe there is a MixinExtra's bug that prevents this Mixin from working:
+     * https://github.com/LlamaLad7/MixinExtras/issues/98
+     * However, I think this is a better Mixin than targeting `itemStack.getItem()`
+     * Let's switch to this Mixin once MixinExtra fixes the bug.
+	@ModifyExpressionValue(at = @At(value = "CONSTANT", args = "classValue=net.minecraft.item.ShieldItem"),
+		method = "blockedByShield(Lnet/minecraft/entity/damage/DamageSource;)V")
+	private boolean instanceOfShieldItem(boolean isVanillaShieldClass) {
+		System.out.println("instanceOfShieldItem");
+		return isVanillaShieldClass;
```

Comment:
I'm assuming this is unfinished, as it just returns the original value, but this injection does seem better than the one below. Once MixinExtras fixes the bug, I agree, let's do it this way.

Reply:
That's why I keep both implementations here. Let's wait for MixinExtras.